plugins/tests/snowflake/test_snowflake.py

```diff
@@ -0,0 +1,76 @@
+from collections import OrderedDict
+
+import pytest
+from flytekitplugins.snowflake import SnowflakeConfig, SnowflakeTask
+
+from flytekit import kwtypes, workflow
+from flytekit.extend import Image, ImageConfig, SerializationSettings, get_serializable
+from flytekit.types.schema import FlyteSchema
+
+query_template = """
+            insert overwrite directory '{{ .rawOutputDataPrefix }}' stored as parquet
+            select *
+            from my_table
+            where ds = '{{ .Inputs.ds }}'
+        """
+
+
+def test_serialization():
+    snowflake_task = SnowflakeTask(
+        name="flytekit.demo.snowflake_task.query",
+        inputs=kwtypes(ds=str),
+        task_config=SnowflakeConfig(
+            account="snowflake", warehouse="my_warehouse", schema="my_schema", database="my_database"
+        ),
+        query_template=query_template,
+        # the schema literal's backend uri will be equal to the value of .raw_output_data
+        output_schema_type=FlyteSchema,
+    )
+
+    @workflow
+    def my_wf(ds: str) -> FlyteSchema:
+        return snowflake_task(ds=ds)
+
+    default_img = Image(name="default", fqn="test", tag="tag")
+    serialization_settings = SerializationSettings(
+        project="proj",
+        domain="dom",
+        version="123",
+        image_config=ImageConfig(default_image=default_img, images=[default_img]),
+        env={},
+    )
+
+    task_spec = get_serializable(OrderedDict(), serialization_settings, snowflake_task)
+    print(task_spec)
```

Comment:
cc @wild-endeavor 
I encountered a strange error when serializing the snowflake task.
pyflyte --pkgs myapp.workflows package --image docker.io/pingsutw/myflyteapp:v1 --fast --forc
```
flytekit/models/common.py:52: in __str__
    return self.verbose_string()
flytekit/models/common.py:67: in verbose_string
    return self.short_string()
flytekit/models/common.py:61: in short_string
    return _six.text_type(self.to_flyte_idl())
flytekit/models/task.py:492: in to_flyte_idl
    return _admin_task.TaskSpec(template=self.template.to_flyte_idl())
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <[AttributeError('to_flyte_idl') raised in repr()] TaskTemplate object at 0x7fa808fe5150>

    def to_flyte_idl(self):
        """
        :rtype: flyteidl.core.tasks_pb2.TaskTemplate
        """
        task_template = _core_task.TaskTemplate(
            id=self.id.to_flyte_idl(),
            type=self.type,
            metadata=self.metadata.to_flyte_idl(),
            interface=self.interface.to_flyte_idl(),
            custom=_json_format.Parse(_json.dumps(self.custom), _struct.Struct()) if self.custom else None,
            container=self.container.to_flyte_idl() if self.container else None,
            task_type_version=self.task_type_version,
            security_context=self.security_context.to_flyte_idl() if self.security_context else None,
            config={k: v for k, v in self.config.items()} if self.config is not None else None,
            k8s_pod=self.k8s_pod.to_flyte_idl() if self.k8s_pod else None,
>           sql=self.sql.to_flyte_idl() if self.sql else None,
        )
E       AttributeError: to_flyte_idl

flytekit/models/task.py:447: AttributeError

```
I already implement `to_flyte_idl` for the SQL class, you can check the code above. 

We can run the test in `test_snowflake.py` to reproduce the error:
```
git clone https://github.com/pingsutw/flytekit.git
cd flytekit
git checkout snowflake-1
pip install .
pip install plugins/flytekit-snowflake
cd ..
git clone https://github.com/pingsutw/flyteidl.git
cd flyteidl
git checkout sql-abstraction
pip install .
cd ..
cd flytekit
pytest plugins/tests/snowflake
```
Error message:
```
elf = <[AttributeError('to_flyte_idl') raised in repr()] TaskTemplate object at 0x7fa808fe5150>

    def to_flyte_idl(self):
        """
        :rtype: flyteidl.core.tasks_pb2.TaskTemplate
        """
        task_template = _core_task.TaskTemplate(
            id=self.id.to_flyte_idl(),
            type=self.type,
            metadata=self.metadata.to_flyte_idl(),
            interface=self.interface.to_flyte_idl(),
            custom=_json_format.Parse(_json.dumps(self.custom), _struct.Struct()) if self.custom else None,
            container=self.container.to_flyte_idl() if self.container else None,
            task_type_version=self.task_type_version,
            security_context=self.security_context.to_flyte_idl() if self.security_context else None,
            config={k: v for k, v in self.config.items()} if self.config is not None else None,
            k8s_pod=self.k8s_pod.to_flyte_idl() if self.k8s_pod else None,
>           sql=self.sql.to_flyte_idl() if self.sql else None,
        )
E       AttributeError: to_flyte_idl

flytekit/models/task.py:447: AttributeError

```
The error only occurred when print task_spec.
The test will pass when commenting this line.